### DISCUSSION TITLE
Query the latest 10 posts of a list of actors

### DISCUSSION THREAD

**Discussion by seboslaw** [bluesky-social/atproto#1683] [How to do it?]
Hey guys,

I'm working on an app and I'd like my users to have the possibility to view only the posts of a list of users. Normally, we'd have to create a list and then possibly a feed of that list to do that. But then the list would be public and I have a huge amount of users that don't want that. 
So my thinking was to use the dids of actors and query the Bluesky API for e.g. the latest 10 posts of each. But I fear there isn't a way to do that right now, is there?

**Comment by snarfed** [comment]
Sure, that's the `getAuthorFeed` XRPC: https://github.com/bluesky-social/atproto/discussions/1662#discussioncomment-7117734

**Reply by seboslaw** [reply]
@snarfed but that will only let me query the feed one author at a time, correct? That would be kinda inefficient, especially if the list of actors is long.

**Reply by snarfed** [reply]
Right. Alternatively, you could make a proxy account that follows the list of users and then fetch that account's timeline. Otherwise though, afaik there isn't yet the "list" functionality you want.

**Reply by dholms** [reply]
we actually do have list functionality now! it's not in the app yet (WIP!) so we do reserve the right to tweak it till then 😛 

https://github.com/bluesky-social/atproto/blob/main/lexicons/app/bsky/feed/getListFeed.json

**Reply by dholms** [reply]
The list does need to be public, but if you don't want it associated with your users, my recommendation would be to create the list through a different "proxy" account. A user can query a "list feed" from a list created by another repo

**Reply by seboslaw** [reply]
Hm, that would require every user to create a proxy account if they don't want everybody to know who e.g. their favourite accounts are that they want to read more frequently - not really feasible :( 
Can't you "just" make app.bsky.feed.getAuthorFeed accept an array of at-identifiers instead of just one? :)

**Reply by dholms** [reply]
You could have a single proxy account for every list from users of your app!

> Can't you "just" make app.bsky.feed.getAuthorFeed accept an array of at-identifiers instead of just one? :)

Unfortunately no. For one, those at-identifiers get passed in query params so it would cap out at a relatively small size. And for larger lists, that's a dynamic query that can be tricky to service. Relational databases struggle to do dynamic queries like that at scale. We're planning on moving to fan-out on write for timelines/feeds which wouldn't work with a dynamic list of uris. 

Maybe in the future we (or someone else!) can build an appview route that does so. But for now, `getListFeed` is close enough to the desired functionality that we're gonna stick with that

**Reply by seboslaw** [reply]
I understand and that totally makes sense. Thank you for the explanation. I'll probably implement it through `getListFeed` and put in a disclaimer that those lists will be public. 
As for private lists: Having a single proxy account for all lists seems like an intermediate solution indeed! However, it's a risky business ;) If that account gets flagged/axed for whatever reason all users will lose access to their data. I'm guessing there's no way to have an account `protected` (and in turn the account maybe cannot interact/post/comment e.g.), right?
